lib/waterline/utils/normalize.js

```diff
@@ -50,6 +50,25 @@ var normalize = module.exports = {
       else if (context.attributes[pk].type == 'string') {
         coercePK = function(pk) {return String(pk).toString();};
       }
+
+      // If the the data type is uuid (both according to the model definition
+      // as well as the database schema) attempt to coerce the value to a
+      // properly formatted uuid, stripping any extreneous characters from it.
+      else if (context.attributes[pk].type == 'uuid' &&
+               context.schema &&
+               context.schema[pk] &&
+               context.schema[pk].type == 'uuid') {
+        coercePK = function(pk) {
+          var pattern = /[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}/i
```

Comment:
Might be better to pull this to the top of the file so we don't need to redeclare the regex every time we match.